src/components/TodoComponent.tsx

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { useState, useEffect } from 'react';
+import { changeTodo } from '../api/todos';
+
+interface TodoComponentProps {
+  todo: Todo;
+  isLoading: boolean;
+  tempTodo: Todo | null;
+  inputRef: React.MutableRefObject<HTMLInputElement | null>;
+  onTodoDelete: (todoId: number) => void;
+  onTodoToggle: (todoId: number, currentCompletedStatus: boolean) => void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onUpdateTodo: any;
```

Comment:
```suggestion
  onUpdateTodo: Promise<Todo>;
```

Wouldn't that be better? Or `Promise<unknown>` at least